File: tl/src/main/java/com/github/badoualy/telegram/tl/api/storage/TLFileMp4.java
package com.github.badoualy.telegram.tl.api.storage;

/**
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
public class TLFileMp4 extends TLAbsFileType {

    public static final int CONSTRUCTOR_ID = 0xb3cea0e4;

    private final String _constructor = "storage.fileMp4#b3cea0e4";

    public TLFileMp4() {
    }

    @Override
    public String toString() {
        return _constructor;
    }

    @Override
    public int getConstructorId() {
        return CONSTRUCTOR_ID;
    }
}

File: tl/src/main/kotlin/com/github/badoualy/telegram/tl/api/storage/TLFileMp4.kt
package com.github.badoualy.telegram.tl.api.storage

/**
 * storage.fileMp4#b3cea0e4
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
class TLFileMp4 : TLAbsFileType() {
    private val _constructor: String = "storage.fileMp4#b3cea0e4"

    override val constructorId: Int = CONSTRUCTOR_ID

    override fun toString() = _constructor

    override fun equals(other: Any?): Boolean {
        if (other !is TLFileMp4) return false
        if (other === this) return true

        return true
    }
    companion object  {
        const val CONSTRUCTOR_ID: Int = 0xb3cea0e4.toInt()
    }
}
